lib/ecmascript.ts

```diff
@@ -140,12 +140,16 @@ export function ToPositiveInteger(valueParam: unknown, property?: string): numbe
   return value;
 }
 
-function ToIntegerNoFraction(valueParam: unknown): number {
+export function ToIntegerWithoutRounding(valueParam: unknown): number {
   const value = ToNumber(valueParam);
+  if (NumberIsNaN(value)) return 0;
```

Comment:
@ptomato, why is `NaN` treated as zero in this context?  I'm not questioning the decision just curious.

Reply:
The only reason is because [ToIntegerOrInfinity](https://tc39.es/ecma262/#sec-tointegerorinfinity) also does that.